lib/Runtime/Language/AsmJsLink.cpp

```diff
@@ -112,19 +112,22 @@ namespace Js{
                 return false;
             }
         }
-        Var asmMathObject = JavascriptOperators::OP_GetProperty(stdlib, PropertyIds::Math, scriptContext);
-        for (int i = 0; i < AsmJSMathBuiltinFunction::AsmJSMathBuiltin_COUNT; i++)
+        if (!mathBuiltinUsed.IsAllClear())
         {
-            //check if bit is set
-            if (!mathBuiltinUsed.Test(i) || i == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_infinity || i == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_nan)
+            Var asmMathObject = JavascriptOperators::OP_GetProperty(stdlib, PropertyIds::Math, scriptContext);
+            for (int i = 0; i < AsmJSMathBuiltinFunction::AsmJSMathBuiltin_COUNT; i++)
             {
-                continue;
-            }
-            AsmJSMathBuiltinFunction mathBuiltinFunc = (AsmJSMathBuiltinFunction)i;
-            if (!CheckMathLibraryMethod(scriptContext, asmMathObject, mathBuiltinFunc))
-            {
-                AsmJSCompiler::OutputError(scriptContext, _u("Asm.js Runtime Error : Math builtin function is invalid"));
-                return false;
+                //check if bit is set
+                if (!mathBuiltinUsed.Test(i) || i == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_infinity || i == AsmJSMathBuiltinFunction::AsmJSMathBuiltin_nan)
```

Comment:
Since you've previously cleared nan and infinity, shouldn't `mathBuiltinUsed.Test(i)` always be false for those 2, thus not requiring to check explicitly for them